src/libs/ReportActionsUtils.ts

```diff
@@ -214,15 +214,53 @@ function getSortedReportActions(reportActions: ReportAction[] | null, shouldSort
         return (first.reportActionID < second.reportActionID ? -1 : 1) * invertedMultiplier;
     });
 
-    // If shouldMarkTheFirstItemAsNewest is true, label the first reportAction as isNewestReportAction
-    if (shouldMarkTheFirstItemAsNewest && sortedActions?.length > 0) {
-        sortedActions[0] = {
-            ...sortedActions[0],
-            isNewestReportAction: true,
-        };
+    return sortedActions;
+}
+
+/**
+ * Returns the range of report actions from the given array which include current id
+ * the range is consistent
+ *
+ * param {ReportAction[]} array
+ * param {String} id
+ * returns {ReportAction}
+ */
+function getRangeFromArrayByID(array: ReportAction[], id?: string): ReportAction[] {
```

Comment:
```suggestion
function getRangeFromArrayByID(sortedReportActions: ReportAction[], id?: string): ReportAction[] {
```

Reply:
Also, I think we need a more descriptive name for this function. Naming is hard, but here are some ideas:

- `getConnectedReportActions`
- `getContinuousReportActionChain`
- `getGaplessReportActions`
- `getLargestGaplessArrayOfReportActionsIncludingID`

Reply:
Probably my favorite is `getContinuousReportActionChain`, because it captures the purpose of the function to detect gaps and return the longest continuous (i.e: "gapless") set of those reportActions. And "chain" highlights that it's essentially a linked list.

Basically we're taking a disjoint linked list that may have gaps, choosing an item from that list, and returning the biggest portion of that list that doesn't have any gaps.

Reply:
`getContinuousReportActionChain` and `getLargestGaplessArrayOfReportActionsIncludingID` sounds more descriptive.
I'll pick `getContinuousReportActionChain` as well
I'll update it now, thanks!